saleor/tax/utils.py

```diff
@@ -248,3 +249,24 @@ def get_shipping_tax_class_kwargs_for_order(tax_class: Optional["TaxClass"]):
         "shipping_tax_class_private_metadata": tax_class.private_metadata,
         "shipping_tax_class_metadata": tax_class.metadata,
     }
+
+
+def check_negative_values_in_tax_data(tax_data: TaxData) -> bool:
+    if (
+        tax_data.shipping_price_gross_amount < 0
+        or tax_data.shipping_price_net_amount < 0
+    ):
+        return True
+
+    for line in tax_data.lines:
+        if line.total_gross_amount < 0 or line.total_net_amount < 0:
+            return True
+
+    return False
+
+
+def check_negative_values_in_tax_data_from_plugin(tax_data: dict[str, Any]) -> bool:
```

Comment:
Similar here

Reply:
The details will be visible in logs attribute